src/marlin_stubs/host/M115.cpp

```diff
@@ -43,9 +43,13 @@ void GcodeSuite::M115() {
     SERIAL_ECHOPGM("FIRMWARE_NAME:Prusa-Firmware-Buddy ");
     SERIAL_ECHOPGM(project_version_full);
     SERIAL_ECHOPGM(" (Github) SOURCE_CODE_URL:https://github.com/prusa3d/Prusa-Firmware-Buddy");
-#if PRINTER_IS_PRUSA_MK4
+#ifdef PRINTER_IS_PRUSA_MK4 || PRINTER_IS_PRUSA_MK4
     SERIAL_ECHOLNPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:"
+    #if PRINTER_IS_PRUSA_MK4
                      "MK4"
+    #elif PRINTER_IS_PRUSA_MK3_5
+                     "MK3.5"
+    #endif
                      " EXTRUDER_COUNT:" STRINGIFY(1) " UUID:" MACHINE_UUID); // extruder hardcoded due to mmu, should be dynamic in the future
```

Comment:
* For XL, this needs to be read dynamically somewhere from the variables to see how many extruders are actually connected
* Otherwise, check against HAS_MMU2() and if so, report 1
* Otherwise, report `EXTRUDERS`

Reply:
Solved.